documentation/book/con-cluster-operator.adoc

```diff
@@ -0,0 +1,21 @@
+[id='cluster-operator-{context}']
+= Overview of Cluster Operator
+
+The Cluster Operator is in charge of deploying a Kafka cluster alongside a Zookeeper ensemble. As part of the Kafka cluster, it can also deploy the Topic Operator which provides operator-style topic management via ConfigMaps. 
+
+ifdef::Kubernetes[]
+The Cluster Operator can also deploy a Kafka Connect cluster which connects to an existing Kafka cluster. 
+endif::Kubernetes[]
+
+On OpenShift, you can create a cluster and connect to an existing Kafka cluster using the Source2Image feature, which provides an easy way of including more connectors.
+
+.Example Architecture diagram of the Cluster Operator.
+image::cluster_operator.png[Cluster Operator]
+
+When the Cluster Operator is up, it _watches_ for certain {ProductPlatformName} resources containing the desired Kafka or Kafka Connect cluster configuration. A `Kafka` resource is used for the Kafka cluster configuration, and a `KafkaConnect` resource is used for Kafka Connect cluster configuration.
+
+When a new desired resource like `Kafka` or `KafkaConnect` resource is created in the {ProductPlatformName} cluster, the operator gets the cluster configuration of the desired resource and starts creating a new Kafka or Kafka Connect cluster by creating the necessary other {ProductPlatformName} resources, such as StatefulSets, Services, ConfigMaps, and so on.
+
+Every time the desired resource is updated by the user, the operator performs corresponding updates on the {ProductPlatformName} resources which make up the Kafka or Kafka Connect cluster. Resources are either patched or deleted and then recreated in order to make the Kafka or Kafka Connect cluster reflect the state of the desired cluster resource. This might cause a rolling update and may lead to service disruption.
```

Comment:
When the resource is _deleted and then recreated_ it will most probably cause deletion of the cluster because the operator could be quite fast at noticing the deletion of the resource. I think we should say here _Resources are either patched or edited in order to_ or if the _edited_ word is too ugly we can probably leave there just _patched_.